test/native/upstream/test_squin2native.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+from kirin.analysis import callgraph
+
+from bloqade import squin
+from bloqade.squin import clifford
+from bloqade.pyqrack import StackMemorySimulator
+from bloqade.native.dialects import gates
+from bloqade.native.upstream import SquinToNative
+
+
+def test_ghz():
+
+    n = 8
+
+    @squin.kernel
+    def main():
+        q = squin.qubit.new(n)
+        squin.h(q[0])
+
+        for i in range(n - 1):
+            squin.cx(q[i], q[i + 1])
+
+        squin.broadcast.sqrt_x_adj(q)
+
+    new_main = SquinToNative().emit(main)
```

Comment:
```suggestion
    new_main = SquinToNative().emit(main, no_raise=False)
```

Just to be sure, I'd always set that in tests.